### PR TITLE
support for console log-levels

### PR DESCRIPTION
**Pull request by timeraider4u**
with these code changes it is possible to write log messages with different severity levels to a console. The log-level of the console can be configured at runtime through eclipse preferences under Window->Preferences->ANTLR4->General->Log-Level. Therefore, the need to set a boolean variable/constant to true or false at compile time is not necessary any more. The console can be injected in ALL Java or Xtend classes of the UI-module:
`@Inject Console console;` and used with, e.g., `console.debug("some message with value='%s'", someVar);`
prefs

### CODE REVIEW

**Comment by HSorensen** [comment]
I like this.
Can you add test cases that tests this ?

File: antlr4ide.core/src/main/java/com/github/jknack/antlr4ide/console/Console.xtend [hunk 1]
```diff
@@ -1,38 +0,0 @@
-package com.github.jknack.antlr4ide.console
```
Comment (HSorensen) [inline antlr4ide.core/src/main/java/com/github/jknack/antlr4ide/console/Console.xtend]: Yeah!! Why have this as an xtend type at all ?

Comment (timeraider4u) [inline antlr4ide.core/src/main/java/com/github/jknack/antlr4ide/console/Console.xtend]: That's exactly my point in https://github.com/antlr4ide/antlr4ide/issues/162 ;-).

File: antlr4ide.ui/src/main/java/com/github/jknack/antlr4ide/ui/preferences/GeneralPreferencePage.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
+/*******************************************************************************
+ * Copyright (c) 2011 itemis AG (http://www.itemis.eu) and others.
```
Comment (timeraider4u) [inline antlr4ide.ui/src/main/java/com/github/jknack/antlr4ide/ui/preferences/GeneralPreferencePage.java]: upps, you are right, will adjust this

**Comment by timeraider4u** [comment]
you mean to add tests which check if log-levels are ignored or printed according to the current preference setting log-level? yes, it can be done but will require some more refactoring. I will take a look soon.

**Comment by timeraider4u** [comment]
Hmmm... maybe I will change to Log4J2 (see https://logging.apache.org/log4j/2.x/manual/index.html) in the background ... would maybe make configuration a little bit easier ... I will give it a try.

**Comment by HSorensen** [comment]
ok, let us wait until you have the log4j2 ready.

**Comment by timeraider4u** [comment]
Okay, I have found out that antlr4ide.tests uses log4j-1.2.17.jar. Interesting! I will check if it is possible to upgrade this to log4j version2.

**Comment by HSorensen** [comment]
@timeraider4u I am tempted to merge this unless you are working on the log4j or log4j2 integration.

**Comment by timeraider4u** [comment]
please do not merge - as I am currently working on implementing the same functionality with log4j - version 1.2.17 - but there are still some issues which I have to resolve locally before I can create a pull request

I will close this pull request now!

**Pull request closed without merging**